### PR TITLE
Mongo and Selenium

### PR DESCRIPTION
**Pull request by Ewalddav**
Cleanup and fix MongoDB connection open/close, reconnect attempts.
Cleanup unused stuff.
Introduce a selenium boolean which will enable crawler and scraper to use selenium to first load the page then pass the HTML to our scraper.

### CODE REVIEW

File: api/edesiaApi.py [hunk 1]
```diff
@@ -1,44 +1,29 @@
...
+api = Api(app)
 
 class Recipe(Resource):
     def get(self, recipe_id):
-        client = pymongo.MongoClient('mongodb://localhost:27017/')
-        db = client.edesia
-        recipe = db.dirty_recipes.find_one({"_id": ObjectId(recipe_id)})
-        client.close()
-        return dumps(recipe, default=my_handler)
+        recipe = mongo.db.dirty_recipes.find_one({"_id": ObjectId(recipe_id)})
```
Comment (jpharroch) [inline api/edesiaApi.py:16]: do we want to use find_one_or_404 here ?

File: api/edesiaApi.py [hunk 2]
```diff
@@ -1,44 +1,29 @@
...
-    elif isinstance(x, ObjectId):
-        return str(x)
-    else:
-        raise TypeError(x)
+app.config['MONGO_URI'] = 'mongodb://localhost:27017/edesia'
+
+mongo = PyMongo(app)
+api = Api(app)
 
 class Recipe(Resource):
```
Comment (jpharroch) [inline api/edesiaApi.py:14]: what about adding decorator @app.route("/recipe/recipeid")

Comment (jpharroch) [reply api/edesiaApi.py:14]: also add , methods=["Get"])

File: api/edesiaApi.py [hunk 3]
```diff
@@ -1,44 +1,29 @@
...
     def put(self):
         recipe = loads(request.data)
-        client = pymongo.MongoClient('mongodb://localhost:27017/')
-        db = client.edesia
-        recipe_id = db.dirty_recipes.insert_one(recipe).inserted_id
-        client.close()
+        recipe_id = mongo.db.dirty_recipes.insert_one(recipe).inserted_id
         return {"recipeId" : str(recipe_id)}
 
 api.add_resource(Recipes, '/recipes')
```
Comment (jpharroch) [inline api/edesiaApi.py:29]: what this does ? if it is true for Recipes what about Recipe ?

File: modules/Core/scraper.py [hunk 4]
```diff
@@ -1,27 +1,31 @@
-import datetime
+from api.edesiaApi import Recipe
```
Comment (jpharroch) [inline modules/Core/scraper.py:1]: move this with the from please :)

**Review by jpharroch — APPROVED** [review]
i have few comments but nothing major. this look really good. I like the changes especially the logic around the IsRecipe

File: api/edesiaApi.py (hunk 1 above)
Comment (Ewalddav) [reply api/edesiaApi.py:16]: Yes!

Comment (Ewalddav) [reply api/edesiaApi.py:16]: Maybe

File: api/edesiaApi.py (hunk 2 above)
Comment (Ewalddav) [reply api/edesiaApi.py:14]: addressed

File: api/edesiaApi.py (hunk 3 above)
Comment (Ewalddav) [reply api/edesiaApi.py:29]: its there

File: modules/Core/scraper.py (hunk 4 above)
Comment (Ewalddav) [reply modules/Core/scraper.py]: Good eye!

**Comment by Ewalddav** [comment]
Still need to add Selenium support before I merge

**Pull request merged**